src/cider/exec/nextgen/context/RuntimeContext.cpp

```diff
@@ -49,4 +53,57 @@ void RuntimeContext::instantiate(const CiderAllocatorPtr& allocator) {
 
   string_heap_ptr_ = std::make_shared<StringHeap>(allocator);
 }
+
+Batch* RuntimeContext::getNonGroupByAggOutputBatch() {
+  AggExprsInfoVector info = reinterpret_cast<CodegenContext::AggBufferDescriptor*>(
+                                buffer_holder_.back().first.get())
+                                ->info_;
+  int8_t* buf = buffer_holder_.back().second->getBuffer();
+  Batch* batch = batch_holder_.back().second.get();
+
+  // allocate mem
+  // row struct
+  auto arrowarray = batch->getArray();
```

Comment:
seems we can wrap a function to create a struct batch for one line or many lines.